### PR TITLE
Add "only" to file module

### PR DESCRIPTION
**Pull request by timofeymukha**
Just a small improvement according to our guidelines.

### CODE REVIEW

**Review by njansson — CHANGES_REQUESTED** [review]
See comments

File: src/io/file.f90 [hunk 1]
```diff
@@ -32,20 +32,26 @@
...
+  use vtk_file, only : vtk_file_t
+  use stl_file, only : stl_file_t
+  use csv_file, only : csv_file_t
   implicit none
 
+  !> A wrapper around a polymorphic `generic_file_t` that handles its init.
+  !! This is essentially a factory for `generic_file_t` descendants additionally
+  !! handling special CSV file parameters (header and precision).
+  !! @remark We could perhaps make a more typical design, with just a factory
+  !! subroutine, like for most other polymorphic types.
```
[the remark this answers is not in the document]
Comment (timofeymukha) [reply src/io/file.f90:54]: Isn't the `init` for `file_t` essentially a factory for `generic_file_t`? That *could* just be a standalone subroutine like for the solver, simcomps, etc., no?

[the remark this answers is not in the document]
Comment (njansson) [reply src/io/file.f90:54]: Yes kind of, but separation here allows us to avoid circular dependencies

**Comment by timofeymukha** [comment]
I accidently pushed something else to this branch :-). I will reopen with the @remark removed.

**Pull request closed without merging**